crates/autopilot/src/domain/settlement/transaction.rs

```diff
@@ -0,0 +1,73 @@
+//! Domain types for on-chain transactions that settle settlements.
+
+use {
+    super::Settlement,
+    crate::{
+        domain::{auction, eth},
+        infra,
+    },
+};
+
+/// A transaction that settles a settlement. Interacts with the settlement
+/// contract `settle` function.
+pub struct Tx {
+    settlement: Settlement,
+    transaction: Transaction,
+    #[allow(dead_code)]
+    receipt: Receipt,
+}
+
+impl Tx {
+    pub async fn new(tx: eth::TxId, eth: infra::Ethereum) -> Result<Self, Error> {
+        let (transaction, receipt) =
+            tokio::try_join!(eth.transaction(tx), eth.transaction_receipt(tx),)?;
+
+        let domain_separator = eth.contracts().settlement_domain_separator();
+        let settlement = Settlement::new(&transaction.input.0.clone().into(), domain_separator)?;
+        Ok(Self {
+            settlement,
+            transaction,
+            receipt,
+        })
+    }
+
+    pub fn auction_id(&self) -> auction::Id {
+        self.settlement.auction_id()
+    }
+
+    pub fn solver(&self) -> eth::Address {
+        self.transaction.solver
+    }
+}
+
+/// An on-chain transaction that settles a settlement.
+#[derive(Debug)]
+pub struct Transaction {
+    /// The hash of the transaction.
+    pub hash: eth::TxId,
+    /// The address of the solver that submitted the transaction.
+    pub solver: eth::Address,
+    /// The call data of the transaction.
+    pub input: eth::Calldata,
+}
+
+/// A receipt of a transaction that settles a settlement.
```

Comment:
nit: "that settled a function"
Otherwise it sounds like it might happen in the future which is not the case.